### PR TITLE
add AutoLock class

### PR DESCRIPTION
**Pull request by muleisheng**

### CODE REVIEW

**Comment by Capri2014** [comment]
@muleisheng Please rebase and clean and commit history

**Comment by jinghaomiao** [comment]
@muleisheng please resolve the conflict.

File: modules/common/util/auto_lock.h [hunk 1]
```diff
@@ -0,0 +1,88 @@
+/******************************************************************************
+ * Copyright 2018 The Apollo Authors. All Rights Reserved.
```
Comment (xiaoxq) [inline modules/common/util/auto_lock.h]: 2019

Comment (muleisheng) [reply modules/common/util/auto_lock.h]: OK, thank you for your remind. I will fix this issue later.

**Comment by muleisheng** [comment]
> @muleisheng please resolve the conflict.

Sorry for delay reply. I will fix this issue later.

**Pull request closed without merging**